lib/matplotlib/transforms.py

```diff
@@ -2791,6 +2791,13 @@ def nonsingular(vmin, vmax, expander=0.001, tiny=1e-15, increasing=True):
         vmin, vmax = vmax, vmin
         swapped = True
 
+    # Expand vmin, vmax to at least a floating point type: if they were integer
+    # types, they can wraparound in abs (abs(np.int8(-128)) == -128) and
+    # vmax - vmin can overflow.
+    vmin, vmax = map(
+        np.promote_types(np.array([vmin, vmax]).dtype, np.float32).type,
```

Comment:
Any particular reason, you are not going to the common float64? These are just two numbers.

Reply:
There are some places where we use float32 because color mapping typically doesn't need double precision and we can save a lot of memory by sticking to float32; I was trying to do the same... but actually here the default expander, 1e-15, was clearly chosen close to the float64 epsilon let's just use the usual float(64) here.